src/neuroconv/utils/json_schema.py

```diff
@@ -76,54 +79,56 @@ def get_schema_from_method_signature(class_method: classmethod, exclude: list =
-                        input_schema["properties"].update({param_name: dict(format="directory")})
-            else:
-                raise NotImplementedError(
-                    f"The annotation type of '{param}' in function '{class_method}' is not implemented! "
-                    "Please request it to be added at github.com/catalystneuro/nwb-conversion-tools/issues "
-                    "or create the json-schema for this method manually."
+                    raise ValueError(f"No valid arguments were found in the json type mapping for parameter {param}")
+                num_params = len(set(param_types))
+                conflict_message = (
+                    "Conflicting json parameter types were detected from the annotation! "
+                    f"{param.annotation.__args__} found."
                 )
-            arg_spec = {param_name: dict(type=param_type)}
-            if param.default is param.empty:
-                input_schema["required"].append(param_name)
-            elif param.default is not None:
-                arg_spec[param_name].update(default=param.default)
-            input_schema["properties"] = dict_deep_update(input_schema["properties"], arg_spec)
+                # Normally cannot support Union[...] of multiple annotation types
+                if num_params > 2:
+                    raise ValueError(conflict_message)
+                # Special condition for Optional[...]
+                if num_params == 2 and not args[1] is type(None):  # noqa: E721
+                    raise ValueError(conflict_message)
+                args_spec[param_name]["type"] = param_types[0]
+            else:
+                arg = param.annotation
+                if arg.__name__ in annotation_json_type_map:
+                    args_spec[param_name]["type"] = annotation_json_type_map[arg.__name__]
+                else:
+                    raise ValueError(
+                        f"No valid arguments were found in the json type mapping '{arg}' for parameter {param}"
+                    )
+                if arg == FilePathType:
+                    input_schema["properties"].update({param_name: dict(format="file")})
+                if arg == FolderPathType:
+                    input_schema["properties"].update({param_name: dict(format="directory")})
+        else:
+            raise NotImplementedError(
+                f"The annotation type of '{param}' in function '{class_method}' is not implemented! "
+                "Please request it to be added at github.com/catalystneuro/nwb-conversion-tools/issues "
+                "or create the json-schema for this method manually."
+            )
+        if param.default is param.empty:
+            input_schema["required"].append(param_name)
+        elif param.default is not None:
+            args_spec[param_name].update(default=param.default)
+        input_schema["properties"] = dict_deep_update(input_schema["properties"], args_spec)
```

Comment:
This was actually a really small change. The large number of lines is due to the fact that I changed

```python
for param_name in params:
    if param_name not in exclude:
          ...
```

to 

```python
for param_name in params:
    if param_name in exclude:
        continue
     ...
```

I prefer this way because it is less indented

Reply:
Thanks for explaining